app/controllers/application_controller.rb

```diff
@@ -177,6 +181,16 @@ def unauthorized e
     head :unauthorized
   end
 
+  def not_implemented e
+    logger.debug "#{e}\n#{e.backtrace.join "\n"}"
+    render json: {
+      error: {
+        code: "not_implemented",
+        message: e.message
+      }
+    }, status: :not_implemented
+  end
```

Comment:
I know you guys are probably just following along with the existing code here, and you should probably leave this as for consistency (we can clean up all of later if we want), but I wanted to point out that the way we are doing this is fragmented.  A better design imo would to make the error objects responsible for their own representation as json, ie, giving the a `to_json` method.  That would clean up all the code like this on this page, enabling us to write `render json: e.to_json`.